### DISCUSSION TITLE
How do i download the source code of my project

### DISCUSSION THREAD

**Discussion by itwela** [vercel/community#5469] [Help] [answered]
### Summary

So i was working on a project locally and deploying to vercel. I hadn’t put it on GitHub because it was sensitive time and i figure i could just download from vercel as i would see i could always view the source code in my deployments so i figured it would always be an option to download as well. But to my surprise, I can’t find one. I’ve tried some third party GitHub repo to download the files but it only downloads 1 file and won’t download anything else. Can you please help me find a way to download the source code I’m so easily able to just pull up in vercel? This seems like a feature that should just be available.

### Steps to Reproduce

I don’t know how you’d recreate this but just go to your deployments and you will see a “source code” button. Click that and that is what I’m trying to download but there no option.

**Comment by jaiveer-y** [comment]
you got the answer my nigga?

**Comment by amyegan** [comment, marked as answer]
For security reasons, we do not provide a way to download source code from Vercel. 

There seem to be a few packages on npm that allow this, but we don't recommend them. Take caution if you choose to use these third party options.
